src/ZPublisher/tests/test_mapply.py

```diff
@@ -90,3 +90,17 @@ class NoCallButAcquisition(Acquisition.Implicit):
 
         ob = NoCallButAcquisition().__of__(Root())
         self.assertRaises(TypeError, mapply, ob, (), {})
+
+    def testFunctionWithSignature(self):
+        from inspect import Parameter
+        from inspect import Signature
+
+        def f(*args, **kw):
+            return args, kw
+
+        f.__signature__ = Signature(
```

Comment:
In the changelog you write about decorators setting `__signature__`. Would it be possible to use such a decorator here to make the test consistent with the change log?

Reply:
Do you think this is necessary?

The change log entry addresses users. Like one of my clients, they likely observe that some methods decorated with help of the `decorator` package can no longer be published correctly. Cause: up to `decorator 4.x`, the package used a complex, Python 2 and `mapply` compatible way to implement signature preserving decorators; `decorator 5.x` dropped Python 2 support and now simply informs about the signature of the decoration wrapper by setting its `__signature__`.

The test checks that `__sigature__` is honored. It is not important how or for what reason it has been set.

If you think, the change log entry and the test must be consistent, I will remove the decorator reference in the change log entry.

Reply:
I do not think adapting the test is absolutely necessary. My idea was to make the test more end-to-end by letting the decorator generate the `__signature__` instead of doing it manually. It seems that to achieve this goal a dependency on the `decorator` package is needed.

So let's keep it like it is, as a new dependency just for a single test case seems too heavy.